### PR TITLE
Add CI job for validating CSV entries

### LINKED ISSUE #142: Check all hyperlinks
We should write a script that checks that all the hyperlinks work.

### PR DESCRIPTION
**Pull request by kandersolar**
Closes #142 and #166.

#142 is about validating the hyperlinks in the CSV file, but it is possible and perhaps useful to extend the validation to other fields as well.  As an example, I have added basic sanity checks on the latitude, longitude, and elevation columns.

The job also adds annotations/comments in the PR on lines that fail the validation checks.

### CODE REVIEW

**Comment by kandersolar** [comment]
It seems to be working well.  One hiccup is that the Australian BOM website rejects automated requests:

> <p>Your access is blocked due to the detection of a potential automated access request. The Bureau of Meteorology website does not support web scraping: if you are trying to access Bureau data through automated means, you should stop. You may like to consider the following options:</p> 
>  
> <ul>
> <li>An anonymous FTP channel: <a href="http://www.bom.gov.au/catalogue/anon-ftp.shtml">http://www.bom.gov.au/catalogue/anon-ftp.shtml</a> -  this is free to access, but use is subject to the default terms of the Bureau's copyright notice: <a href="http://www.bom.gov.au/other/copyright.shtml">http://www.bom.gov.au/other/copyright.shtml</a> </li>
> <li>A Registered User service for continued use of Bureau data if your activity does not comply with the default terms: <a href="http://reg.bom.gov.au/other/charges.shtml">http://reg.bom.gov.au/other/charges.shtml</a> noting charges apply to most data products. Please contact webreg@bom.gov.au to discuss your requirements.</li>
> </ul> 
>  
> <p>If you still need assistance in accessing our website, please contact us by filling in your details at <a href="http://reg.bom.gov.au/screenscraper/screenscraper_enquiry_form/">http://reg.bom.gov.au/screenscraper/screenscraper_enquiry_form/</a> and we will get in touch with you. Thank you for your understanding.</p>

It is presumably easy enough to bypass this by specifying a useragent or somesuch, but perhaps it is better to just comply.  I suppose we could add an exception in the validator to skip certain URLs.  @AdamRJensen?

The other URL failures seem to be legitimate issues.  I suggest addressing them in separate PRs to keep this one focused.

File: .github/workflows/validate-csv.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - name: Install Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: Install requirements
+        run: pip install requests pandas numpy
+
+      - name: Set up output matcher for PR annotations
+        run: echo '::add-matcher::.github/workflows/validate-csv-matcher.json'
```
Comment (AdamRJensen) [inline .github/workflows/validate-csv.yml:27]: Can you explain a bit about what happens here? This doesn't seem like Python 😆

Comment (kandersolar) [reply .github/workflows/validate-csv.yml:27]: This is using a feature of GitHub Actions to enable automatic annotations/comments in the pull requests's file view, like this:

image

Basically:
- Some part of the CI job emits logging output containing information to be used for annotations: https://github.com/AssessingSolar/solarstations/pull/143/files#diff-0dcf874f7ba8617229c0b2282660a0d73587408a21c4ccefac267ea348cc385eR75-R76
- The matcher.json file defines a regex that identifies those special log output lines and extracts the necessary annotation information (which file, which line, what message to display in the annotation): https://github.com/AssessingSolar/solarstations/pull/143/files#diff-7cd4b2d9ed8f98f39437ec6dbc12a17d4fd790a14daf6c6c546f4aca810941beR8-R12
- This `echo '::add-matcher::.github/workflows/validate-csv-matcher.json'` line enables the specified matching+annotations for this CI job

**Comment by AdamRJensen** [comment]
This is awesome @kandersolar, thank you so much!

**Pull request merged**